driver-core/src/main/com/mongodb/client/model/expressions/MqlExpression.java

```diff
@@ -395,6 +401,77 @@ public T reduce(final T initialValue, final BinaryOperator<T> in) {
                 .append("in", extractBsonValue(cr, in.apply(varValue, varThis)))));
     }
 
+    @Override
+    public BooleanExpression any(final Function<? super T, BooleanExpression> predicate) {
+        MqlExpression<BooleanExpression> map = (MqlExpression<BooleanExpression>) this.map(predicate);
```

Comment:
Let's rename the variable `map` here and in 6 more places below to something that is less confusing because `this.map` produces an array and not a map.